knox.go

```diff
@@ -22,10 +24,22 @@ type KnoxClient interface {
 	GenerateIdentity(GenerateIdentityParams) error
 }
 
-func NewKnoxClient(s signer.DynamicSigner) (KnoxClient, error) {
-	ca, err := credential_adapter.NewCredentialAdapterClient()
+type KnoxConfig struct {
+	Signer signer.DynamicSigner
+	Issuer struct {
+		CredentialAdapterURL string
+		AuthServiceURL       string
+	}
+}
+
+func NewKnoxClient(c KnoxConfig) (KnoxClient, error) {
```

Comment:
Mabe it is better to declare dependencies explicitly and pass it to the constructor so that NewKnoxClient() accepted credentialAdapterClient and authClient rather than KnoxConfig?

Reply:
The purpose of the SDK and `NewKnoxClient` is to abstract away the network level stuff. We don't want the SDK consumer to have to create the clients, that defeats the purpose of the SDKs. And we also don't need to add them to the `NewKnoxClient` function as we can just create the struct for testing and then put the mocks there.